Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaWorkSpace.java

```diff
@@ -23,33 +23,158 @@
+
+    @NotNull
+    private final String name;
+
+    @NotNull
+    private final AzureHttpObservable http;
+
+    public ArcadiaWorkSpace(@NotNull SubscriptionDetail subscription, @NotNull Workspace workspaceResponse) {
+        this.subscription = subscription;
+        this.workspaceResponse = workspaceResponse;
+        this.name = workspaceResponse.name();
+        this.uri = URI.create(CommonSettings.getAdEnvironment().resourceManagerEndpoint()).resolve(workspaceResponse.id());
+        this.http = new AzureHttpObservable(subscription, ApiVersion.VERSION);
+    }
 
     @NotNull
     @Override
     public ImmutableSortedSet<? extends IClusterDetail> getClusters() {
-        return ImmutableSortedSet.of();
+        return ImmutableSortedSet.copyOf(
+                this.clusters.stream().filter(cluster -> ((ArcadiaSparkCompute) cluster).isRunning()).iterator());
     }
 
     @NotNull
     public Observable<ArcadiaWorkSpace> fetchClusters() {
-        return Observable.just(this);
+        return getSparkComputesRequest()
+                .map(this::updateWithResponse)
+                .defaultIfEmpty(this);
+    }
+
+    @NotNull
+    private Observable<GetSparkComputeListResponse> getSparkComputesRequest() {
+        String url = getUri().toString() + REST_SEGMENT_SPARK_COMPUTES;
+
+        return getHttp()
+                .withUuidUserAgent()
+                .get(url, null, null, GetSparkComputeListResponse.class);
+    }
+
+    private ArcadiaWorkSpace updateWithResponse(@NotNull GetSparkComputeListResponse response) {
+        this.clusters =
+                ImmutableSortedSet.copyOf(response.items().stream()
+                        .map(sparkCompute -> new ArcadiaSparkCompute(this, sparkCompute))
+                        .iterator());
+        return this;
     }
 
     @NotNull
     @Override
     public ClusterContainer refresh() {
-        return this;
+        try {
+            return fetchClusters().toBlocking().singleOrDefault(this);
+        } catch (Exception ignored) {
+            log().warn("Got Exceptions when refreshing Arcadia spark computes. " + ExceptionUtils.getStackTrace(ignored));
+            return this;
+        }
+    }
+
+    public boolean isRunning() {
+        if (getProvisioningState() == null) {
+            return false;
+        }
+
+        return getProvisioningState().equals(WorkspaceProvisioningState.PROVISIONING)
+                || getProvisioningState().equals(WorkspaceProvisioningState.SUCCEEDED);
+    }
+
+    @Nullable
+    public WorkspaceProvisioningState getProvisioningState() {
+        if (this.workspaceResponse.properties() == null) {
+            return null;
+        }
+
+        return this.workspaceResponse.properties().provisioningState();
+    }
+
+    @NotNull
+    public String getState() {
+        return Optional.ofNullable(getProvisioningState()).map(state -> state.toString()).orElse("Unknown");
+    }
+
+    @NotNull
+    public String getTitleForNode() {
```

Comment:
>getTitle [](start = 18, length = 8)

If cluster is in `Succeeded` provisioning state, we should avoid showing such state, because this state might be misleading as job state.